sonar-db/src/main/java/org/sonar/db/user/RoleDao.java

```diff
@@ -40,35 +37,31 @@
     return session.getMapper(RoleMapper.class).selectGroupPermissions(groupName, resourceId, DefaultGroups.isAnyone(groupName));
   }
 
-  public void insertGroupRole(DbSession session, GroupRoleDto groupRole) {
-    mapper(session).insertGroupRole(groupRole);
-  }
-
-  public void insertUserRole(DbSession session, UserPermissionDto userRole) {
-    mapper(session).insertUserRole(userRole);
+  public void insertGroupRole(DbSession session, GroupPermissionDto dto) {
+    mapper(session).insertGroupRole(dto);
   }
 
-  public void deleteUserRole(UserPermissionDto userRole, SqlSession session) {
-    mapper(session).deleteUserRole(userRole);
+  /**
+   * @deprecated replaced by {@link org.sonar.db.permission.UserPermissionDao#insert(DbSession, org.sonar.db.permission.UserPermissionDto)}
+   */
+  @Deprecated
+  public void insertUserRole(DbSession session, UserPermissionDto dto) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)